parser/src/parser.rs

```diff
@@ -32,6 +32,41 @@ struct ParserState<'a> {
     path: Option<&'a Path>,
 }
 
+fn escape_string (s: &str) -> String {
+    let mut result = String::with_capacity(s.len() - 2);
+
+    let mut chars = s[1 .. s.len()-1].chars();
+
+    while let Some(c) = chars.next() {
+        if c == '\\' {
+            let c = chars.next().expect("Unfinished escape sequence");
+            match c {
+                '\\' => result.push('\\'),
+                '"' => result.push('"'),
+                'n' => result.push('\n'),
+                'r' => result.push('\r'),
+                'x' => {
+                    let a = chars.next()
+                        .expect("Unfinished escape sequence")
+                        .to_digit(16)
+                        .expect("Invalid hexadecimal digit");
+                    let b = chars.next()
+                        .expect("Unfinished escape sequence")
+                        .to_digit(16)
+                        .expect("Invalid hexadecimal digit");
+                    // Fails when not ascii
+                    result.push((a*16 + b) as u8 as char);
+                }
+                c => panic!("Invalid escape sequence: \\{}", c)
+            }
+        } else {
+            result.push(c);
+        }
+    }
+
+    result
+}
```

Comment:
This function will need to return a `Result` seeing the number of things that can panic, unless they are all cases prevented by the parser.